react-app/src/pages/PlotPage.tsx

```diff
@@ -156,11 +168,46 @@ export default function PlotPage(): ReactElement {
     }));
   };
 
-  // Arrays for dropdown menu items
-  let geneDropDownItemsArray = ["IGHV", "IGHD", "IGKJ", "..."];
-  let subtypeDropDownItemsArray = ["1-2", "3-4", "..."];
-  let alleleDropDownItemsArray = ["*02", "*04", "*05", "*06", "..."];
-  let geneSegmentItemsArray = ["IGH", "..."];
+    // fetch next selection options after gene is selected
+    useEffect(() => {
+      if (!currentPicks.subtypeDropdown) {
+        setAlleleDropDownItemsArray(["..."]);
+        let currentSelection = currentPicks.geneDropdown;
+        axios
+          .get(geneSelectionEndpoint + currentSelection)
+          .then((response) => {
+            let responseData = response.data;
+            responseData.push("...");
+            setSubtypeDropDownItemsArray(responseData);
+          })
+          .catch((response) => console.log(response.error));
+      } else {
+        let currentSelection = currentPicks.geneDropdown + currentPicks.subtypeDropdown + '*';
+        setAlleleDropDownItemsArray([])
+        axios
+          .get(geneSelectionEndpoint + currentSelection)
+          .then((response) => {
+            let responseData = response.data;
+            responseData.push("...");
+            setAlleleDropDownItemsArray(responseData);
+          })
+          .catch((response) => console.log(response.error));
+      }
+    }, [currentPicks]);
+
+    // fetch next selection options after gene is selected
+    useEffect(() => {
+      let currentSelection = currentPicks.geneDropdown + currentPicks.geneSegmentDropdown;
+      setAlleleDropDownItemsArray([])
+      axios
+        .get(geneSelectionEndpoint + currentSelection)
+        .then((response) => {
+          let responseData = response.data;
+          responseData.push("...");
+          setAlleleDropDownItemsArray(responseData);
```

Comment:
So we set the allele dropdown items with backend data already when we have picked the gene segment?

Reply:
I accidentally left part of the code in after I did some changes, removed that part. They should both be in the same useEffect but in different if/else conditions.